### DISCUSSION TITLE
x-collapse overflow hidden bug

### DISCUSSION THREAD

**Discussion by fadarrizz** [alpinejs/alpine#3550] [7. Bugs (archived)]
It seems like #3243 introduced a bug in the collapse plugin, regarding `overflow: hidden` sometimes being present on the element when it's not collapsed.

The problem I'm seeing is that `getBoundingClientRect()` occasionally gives inaccurate results.

I've tested the result of `getBoundingClientRect()` by placing a `console.log` in the last callback of the Alpine.transition in the in function. Then, I collapsed the same element multiple times. Here are the results:

**With Tailwind**
![image](https://user-images.githubusercontent.com/22003779/235124845-eeb5249c-7fd2-43e2-9492-7be43122adb7.png)

**Without Tailwind**
![image](https://user-images.githubusercontent.com/22003779/235139874-de6663cb-8452-46d4-99e4-1e19e43b10b3.png)

I've tested this on Firefox, Chrome and Safari. Both Firefox and Safari have the same issue, Chrome does not for some reason.

Tailwind seems to make the issue worse. I've not been able to pin down how this happens.

I'm seeing the issue on https://alpinejs.dev/plugins/collapse as well. So it's not just on my machine.

Pr #3121 seems to fix this bug and the bug that #3243 was trying to solve.

**Comment by joshhanley** [comment]
This issue has been converted to a discussion as we are now using discussions to track bugs and ideas.
See this announcement for more details https://github.com/alpinejs/alpine/discussions/1423
